vertx-web/src/main/java/io/vertx/ext/web/handler/impl/SessionHandlerImpl.java

```diff
@@ -353,6 +354,21 @@ public Session newSession(RoutingContext context) {
     return session;
   }
 
+  public User setUser(RoutingContext context, User user) {
+    if (!cookieless) {
+      context.removeCookie(sessionCookieName, false);
+    }
+    context.setUser(user);
+    // signal we must store the user to link it to the session
+    context.put(SESSION_STOREUSER_KEY, true);
+    flush(context, true, true, flush -> {
```

Comment:
Given that we probably should change the method signature, then this callback isn't discarded anymore.

Reply:
I reworked this to return a future that completes when the flush completes.

Reply:
By using a promise, you can reduce the flush code to just:

```java
flush(context, true, true, promise);
```

The promise implements `Handler<AsyncResult<T>>`